src/activities/components/ActivityList.vue

```diff
@@ -334,9 +334,50 @@ export default {
       icsDialog: false,
       numDisplayed: NUM_ACTIVITIES_PER_LOAD,
       types: [],
+      activityListTypeFilter: this.filterActivityTypes.slice(),
     }
   },
   computed: {
+    activityTypes () {
+      const archivedTypeList = []
+      const archivedTypes = {}
+
+      for (const type of this.activityListTypeFilter) {
+        // populate a list of all activity types locally to avoid side effects
+        archivedTypeList.push(type)
+      }
+
+      for (const activityType of archivedTypeList) {
+        // count the number of archived types and store in a hashmap, defaulting to zero
+        if (activityType.status === 'archived') {
+          archivedTypes[activityType.id] = false
+        }
+      }
+
+      for (const activity of this.activities) {
+        // check to see if each archived type has any activities and update the hashmap
+        if (activity.activityType.id in archivedTypes) {
+          const id = activity.activityType.id
+          archivedTypes[id] = true
+        }
+      }
+
+      for (const archivedTypeId in archivedTypes) {
+        console.log(archivedTypeId)
+        if (archivedTypes[archivedTypeId] === false) {
+          // For any archived types with zero activities in view, remove the type from the filter
+          for (const [index, activityType] of archivedTypeList.entries()) {
+            if (parseInt(activityType.id) === parseInt(archivedTypeId)) {
+              // delete archived type from the filter
+              archivedTypeList.splice(index, 1)
+              break
+            }
+          }
+        }
+      }
+
+      return archivedTypeList
```

Comment:
This variable name seems misleading, it's not a list of archived types... maybe just call it `activityTypes`?